src/pepr/operator/utils/waypoint.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Waypoint utility module for standardized waypoint operations
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { UDSPackage } from "../crd";
+import { Mode } from "../crd/generated/package-v1alpha1";
+// No logger needed in this utility module
+
+// Constants for waypoint configuration
+const WAYPOINT_SUFFIX = "-waypoint"; // Suffix for waypoint resource names
+const WAYPOINT_PREFIX = "uds-core-"; // Prefix for waypoint resource names
+
+// Cache for waypoint names to avoid regenerating them
+const waypointNameCache = new Map<string, string>();
+
+/**
+ * Determines if a package should use ambient waypoint networking
+ * @param pkg - The UDS package to check
+ * @returns boolean indicating if ambient waypoint should be used
+ */
+export const shouldUseAmbientWaypoint = (pkg?: UDSPackage): boolean => {
+  if (!pkg) return false;
+
+  // Check if package has ambient mode and authservice SSO
+  return pkg.spec?.network?.serviceMesh?.mode === Mode.Ambient && hasAuthserviceSSO(pkg);
+};
+
+/**
+ * Checks if a package has authservice SSO configuration
+ * @param pkg - The UDS package to check
+ * @returns boolean indicating if package has authservice SSO
+ */
+export const hasAuthserviceSSO = (pkg?: UDSPackage): boolean =>
+  pkg?.spec?.sso?.some(s => !!s.enableAuthserviceSelector) || false;
```

Comment:
Similar comment here - we should be requiring `pkg`. I'm also concerned that this check might not be quite evaluating authservice SSO presence correctly. `enableAuthserviceSelector: {}` is valid and means they want authservice enabled (just on the whole namespace rather than a specific selector). I think this would evaluate to false though since that is an empty object?

Could reference our filter in the main `authservice` function for how we've been doing this: https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/operator/controllers/keycloak/authservice/authservice.ts#L32